framework/include/dirackernels/ADDiracKernel.h

```diff
@@ -0,0 +1,67 @@
+//* This file is part of the MOOSE framework
+//* https://mooseframework.inl.gov
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "DiracKernelBase.h"
+#include "MooseVariableInterface.h"
+#include "ADFunctorInterface.h"
+
+/**
+ * AD version of DiracKernel
+ */
+class ADDiracKernel : public DiracKernelBase,
+                      public MooseVariableInterface<Real>,
+                      public ADFunctorInterface
+{
+public:
+  static InputParameters validParams();
+
+  ADDiracKernel(const InputParameters & parameters);
+
+  virtual void jacobianSetup() override;
+
+  virtual void computeResidual() override;
+  virtual void computeJacobian() override;
+  virtual void computeOffDiagJacobian(unsigned int jvar) override;
+  virtual void computeResidualAndJacobian() override;
+
+  virtual const MooseVariableField<Real> & variable() const override { return _var; }
+
+protected:
+  /**
+   * Computes the residual contribution at the current quadrature point
+   */
+  virtual ADReal computeQpResidual() = 0;
+
+  /// Variable this kernel acts on
+  MooseVariableField<Real> & _var;
+
+  /// Values of shape functions at QPs
```

Comment:
do Dirac kernel involve the whole element? Or is this a single Qp at the dirac point